packages/core-data/src/selectors.ts

```diff
@@ -1325,14 +1338,50 @@ export function getBlockPatternCategories( state: State ): Array< any > {
 	return state.blockPatternCategories;
 }
 
+/**
+ * Retrieve the list of registered user patterns.
+ *
+ * @param state Data state.
+ *
+ * @return User pattern list.
+ */
+export const getUserPatterns = createSelector(
+	( state: State ): Array< UserPattern > => {
+		const categories = new Map();
+		state.userPatternCategories.forEach( ( userCategory ) =>
+			categories.set( userCategory.id, userCategory )
+		);
+
+		return state.userPatterns.map( ( patternBlock: any ) => ( {
+			blocks: parse( patternBlock.content.raw, {
```

Comment:
Calling `parse` in a selector is potentially a problem because of block lazy loading (see #51778 and #53260). With block lazy loading, parsing content into blocks is an `async` function because the parser internally lazily loads the blocks it encounters in the content.

And selectors are synchronous, unless you want to also add a `getUserPatterns` resolver, so they can't really call async functions.

We'll either need to move content parsing out of the selector, or, at least, mark the `getUserPatterns` selector as experimental/private, so that we're not stuck with it forever.

Reply:
Ah, great to know this, thanks for this feedback, will have another think about the best approach.